pkg/types/baremetal/validation/platform_test.go

```diff
@@ -191,10 +209,84 @@ func TestValidatePlatform(t *testing.T) {
+				ClusterOSImage:          "http://192.168.111.1/images/metal.x86_64.qcow2.gz?sha256=340dfa4d92450f2eee852ed1e2d02e3138cc68d824827ef9cf0a40a7ea2f93da",
+			},
+			network:  network,
+			expected: "The URI provided is invalid",
+		},
+		{
+			name: "invalid_clusterosimage",
+			platform: &baremetal.Platform{
+				APIVIP:                  "192.168.111.2",
+				DNSVIP:                  "192.168.111.3",
+				IngressVIP:              "192.168.111.4",
+				Hosts:                   []*baremetal.Host{},
+				LibvirtURI:              "qemu://system",
+				ClusterProvisioningIP:   "172.22.0.3",
+				BootstrapProvisioningIP: "172.22.0.2",
+				ExternalBridge:          "br0",
+				ProvisioningBridge:      "br1",
+				BootstrapOSImage:        "http://192.168.111.1/images/qemu.x86_64.qcow2.gz?sha256=3b5a882c2af3e19d515b961855d144f293cab30190c2bdedd661af31a1fc4e2f",
+				ClusterOSImage:          "http//192.168.111.1/images/metal.x86_64.qcow2.gz?sha256=340dfa4d92450f2eee852ed1e2d02e3138cc68d824827ef9cf0a40a7ea2f93da",
+			},
+			network:  network,
+			expected: "The URI provided is invalid",
+		},
+		{
+			name: "invalid_bootstraposimage_checksum",
+			platform: &baremetal.Platform{
+				APIVIP:                  "192.168.111.2",
+				DNSVIP:                  "192.168.111.3",
+				IngressVIP:              "192.168.111.4",
+				Hosts:                   []*baremetal.Host{},
+				LibvirtURI:              "qemu://system",
+				ClusterProvisioningIP:   "172.22.0.3",
+				BootstrapProvisioningIP: "172.22.0.2",
+				ExternalBridge:          "br0",
+				ProvisioningBridge:      "br1",
+				BootstrapOSImage:        "http://192.168.111.1/images/qemu.x86_64.qcow2.gz?md5sum=3b5a882c2af3e19d515b961855d144f293cab30190c2bdedd661af31a1fc4e2f",
+				ClusterOSImage:          "http://192.168.111.1/images/metal.x86_64.qcow2.gz?sha256=340dfa4d92450f2eee852ed1e2d02e3138cc68d824827ef9cf0a40a7ea2f93da",
+			},
+			network:  network,
+			expected: "The sha256 parameter in the URI is either invalid or missing",
+		},
+		{
+			name: "invalid_clusterosimage_checksum",
+			platform: &baremetal.Platform{
+				APIVIP:                  "192.168.111.2",
+				DNSVIP:                  "192.168.111.3",
+				IngressVIP:              "192.168.111.4",
+				Hosts:                   []*baremetal.Host{},
+				LibvirtURI:              "qemu://system",
+				ClusterProvisioningIP:   "172.22.0.3",
+				BootstrapProvisioningIP: "172.22.0.2",
+				ExternalBridge:          "br0",
+				ProvisioningBridge:      "br1",
+				BootstrapOSImage:        "http://192.168.111.1/images/qemu.x86_64.qcow2.gz?sha256=3b5a882c2af3e19d515b961855d144f293cab30190c2bdedd661af31a1fc4e2f",
+				ClusterOSImage:          "http://192.168.111.1/images/metal.x86_64.qcow2.gz?sha256=3ee852ed1e2d02e3138cc68d824827ef9cf0a40a7ea2f93da",
+			},
+			network:  network,
+			expected: "The sha256 parameter in the URI is either invalid or missing",
+		},
```

Comment:
i think based on the doc, these should be http/https schemes right? So we should validate that and also have a test for that.

Reply:
Aren't we doing this indirectly using url.ParseRequestURI? But I guess if we wanted to restrict this to http/https only then, I agree, we should be explicity.

Reply:
I would prefer not to be in the business of making assumptions about what protocol a user would use. A variety of them are possible beyond http[s].

For the bootstrap, it may be useful to be able to point to file:/// in some instances. I also don't think it'd be unreasonable to allow ftp

IMHO, it's up to the user to know what protocols work best for them and we should just check that it's a valid URL.

Reply:
Sorry I may be mistaken here - the caching code assumes http[s] so we probably should check the override: https://github.com/RobertKrawitz/installer/blob/d9955c20d5707a11c41930c4df42936d5cf811ac/pkg/tfvars/internal/cache/cache.go

Reply:
I've added the check for http/https in the latest commit.